src/main/java/core/basesyntax/service/impl/FileWriterImpl.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.FileWriter;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FileWriterImpl implements FileWriter {
+    private static final String FIRST_LINE = "fruit,quantity";
+    private static final String COMMA_SEPARATOR = ",";
+
+    @Override
+    public void writeDataToFile(String path) {
+        try (BufferedWriter bufferedWriter = Files.newBufferedWriter(Paths.get(path))) {
+            writeFirstLineToFile(bufferedWriter);
+            Storage.STORAGE_MAP.entrySet().stream()
+                    .map(entry -> entry.getKey() + COMMA_SEPARATOR
+                            + entry.getValue() + System.lineSeparator())
+                    .forEach(line -> {
```

Comment:
Let's move logic that creates report String (or List<String>) to the separate class `List<String> ReportService.createReport()`. Put the generated report data to this class as method parameter `writeDataToFile(String path, List<String> lines) {`